packages/api/src/utils/client/httpClient.ts

```diff
@@ -279,6 +286,14 @@ export class HttpClient implements IHttpClient {
       if (bearerToken && headers["Authorization"] === undefined) {
         headers["Authorization"] = `Bearer ${bearerToken}`;
       }
+      if (url.username || url.password) {
```

Comment:
This should be a compound condition for both, we can't generate a valid Authorization header otherwise.

Reply:
As per RFC (https://datatracker.ietf.org/doc/html/rfc2617) those are optional and can be omitted (`*`, i.e. 0 or more).

```
     user-pass   = userid ":" password
      userid      = *<TEXT excluding ":">
      password    = *TEXT
```

In both cases fetch also throws an error as it does not allow credentials in the URL

```
fetch("http://user@localhost")
TypeError: Request cannot be constructed from a URL that includes credentials: http://user@localhost
```

```
fetch("http://:password@localhost")
> Uncaught:
TypeError: Request cannot be constructed from a URL that includes credentials: http://:password@localhost
```